ballerina/engine.bal

```diff
@@ -259,20 +283,28 @@ isolated class Engine {
 
         (readonly & Interceptor)? interceptor = context.getNextInterceptor('field);
         __Type fieldType = 'field.getFieldType();
-        ResponseGenerator responseGenerator = new (self, context, fieldType, 'field.getPath().clone());
+        ResponseGenerator responseGenerator = new (self, context, fieldType, 'field.getPath().clone(), 'field.getCacheConfig().clone(), 'field.getParentArgHashes().clone());
         do {
             if interceptor is readonly & Interceptor {
                 any|error result = self.executeInterceptor(interceptor, 'field, context);
                 string interceptorName = self.getInterceptorName(interceptor);
                 return check validateInterceptorReturnValue(fieldType, result, interceptorName);
             }
             any fieldValue;
-            if 'field.getOperationType() == parser:OPERATION_QUERY {
-                fieldValue = check self.resolveResourceMethod(context, 'field, responseGenerator);
-            } else if 'field.getOperationType() == parser:OPERATION_MUTATION {
-                fieldValue = check self.resolveRemoteMethod(context, 'field, responseGenerator);
+            if 'field.getOperationType() == parser:OPERATION_QUERY && 'field.isCacheEnabled() {
+                string cacheKey = 'field.getCacheKey();
+                any|error cachedValue = self.getFromCache(cacheKey);
```

Comment:
The field value is `any` type since there are service objects.

Reply:
We don't need to cache the service objects. We should cache the values. Caching objects defeats the purpose of caching as we still have to execute the resource methods, which will result in DB calls, etc.

Reply:
Service object is cached to reduce the execution of resource that return the service object. Since the resources are also cached, do not need to execute its resource method after cached.

Reply:
That makes sense.